containers/src/Data/Graph.hs

```diff
@@ -233,11 +259,7 @@ stronglyConnComp
                 -- nodes of the graph; such edges are ignored.
         -> [SCC node]
 
-stronglyConnComp edges0
-  = map get_node (stronglyConnCompR edges0)
-  where
-    get_node (AcyclicSCC (n, _, _)) = AcyclicSCC n
-    get_node (CyclicSCC triples)     = CyclicSCC [n | (n,_,_) <- triples]
+stronglyConnComp = map (fmap (\(n, _, _) -> n)) . stronglyConnCompR
```

Comment:
This is lazier than what we had before. That inner `fmap` isn't nice.

Reply:
`[n | (n,_,_) <- triples]` is more eager than `List.map (\(n,_,_) -> n) triples`. In some other contexts, list fusion and inlining will turn the latter into the former, but I really don't think that's the case here.